src/charm.py

```diff
@@ -182,6 +214,36 @@ def _apply_k8s_resources(self, force_conflicts: bool = False) -> None:
             else:
                 raise GenericCharmRuntimeError("K8S resources creation failed") from error
 
+        for i in range(0, 3):
+            while True:
+                try:
+                    self.training_runtimes_resource_handler.apply()
+                except ApiError as error:
+                    if self._check_and_report_k8s_conflict(error) and force_conflicts:
+                        # conflict detected when applying TrainingRuntime resources
+                        # re-apply TrainingRuntime resources with forced conflict resolution
+                        self.unit.status = MaintenanceStatus(
+                            "Force applying TrainingRuntime resources"
+                        )
+                        self.logger.warning(
+                            "Applying TrainingRuntime resources with conflict resolution"
+                        )
```

Comment:
question: were you seeing conflicts error? iiuc chisme now applies with force [by default](https://github.com/canonical/charmed-kubeflow-chisme/blob/main/src/charmed_kubeflow_chisme/kubernetes/_kubernetes_resource_handler.py#L295) so this case candling should not be needed.